pkgs/build-support/record-playback/record-command

```diff
@@ -0,0 +1,24 @@
+#! /bin/sh
+
+if [ "$1" = "--playback" ]; then
+  shift
+  PLAYBACK=1
+fi
+
+cmd="$1"; shift
+shortargs="$(printf %s "$*" | tr -d '[:space:]')"
+dir=".record-playback/$cmd/$shortargs"
+
+if [ -n "$PLAYBACK" ]; then
+  printf "playing back \`%s %s' from %s...\n" "$cmd" "$*" "$dir" >&2
```

Comment:
The command error output might be piped into another command, I will recommend to use a separate fd for than, such as `>&3`, and then we can configure the setup script to forward any output to 3rd file descriptor to the standard error.